src/server/plugins/plugin.js

```diff
@@ -127,8 +127,10 @@ module.exports = class Plugin {
         server.exposeStaticDir(`/plugins/${id}/{path*}`, this.publicDir);
       }
 
-      this.status = kbnServer.status.createForPlugin(this);
-      server.expose('status', this.status);
+      if (this.externalInit !== _.noop) {
```

Comment:
Also, I think this merits a comment, to explain the intention.

```javascript
// Many of the plugins are simply adding static assets to the server and we don't need
// to track their "status". Since plugins must have an init() function to even set its status
// we shouldn't even create a status unless the plugin can use it.
```